platform/swift/source/logging/Span.swift

```diff
@@ -27,16 +27,21 @@ public final class Span {
         case end = "end"
     }
 
-    private let timeProvider: TimeProvider
+    /// This is the autogenerated unique identifier of the span
+    public let id = UUID()
```

Comment:
One of the asks was to be able to end a span using solely this ID (I supposed that's why we exposed this?) since they didn't want to rely on having to pass full span objects around but I don't see this functionality added anywhere. I think we should consider adding a `endSpan()` method that just takes this ID (yes, that would mean we would need to persist some state around in-flight spans ourselves)

Reply:
also, this is not exposed on the Android side (not sure if on purpose)